src/App.tsx

```diff
@@ -15,48 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
+interface SortParms {
+  sortField: string,
+  isReversed: boolean,
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortParms,
+) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed
+    ? prepearedGoods.reverse()
+    : prepearedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortField, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.alphabet,
+          })}
+          onClick={() => setSortField(SortType.alphabet)}
```

Comment:
for both sort buttons make one handler factory which will accept SortType and return arrow function
```
const hanldeSort = (sort: SortType) => () => setSortField(sort);
```
```suggestion
          onClick={handleSort(SortType.alphabet)}
```